internal/entclient/devurl.go

```diff
@@ -24,22 +32,56 @@ func (c Client) DelDevURL(envID, urlID string) error {
 	return nil
 }
 
-func (c Client) UpsertDevURL(envID, port, access string) error {
+type createDevURLRequest struct {
+	EnvID  string `json:"environment_id"`
+	Port   int    `json:"port"`
+	Access string `json:"access"`
+	Name   string `json:"name"`
+}
+
+// InsertDevURL inserts a new devurl into the specified environment.
+func (c Client) InsertDevURL(envID string, port int, name, access string) error {
 	reqString := "/api/environments/%s/devurls"
-	reqUrl := fmt.Sprintf(reqString, envID)
+	reqURL := fmt.Sprintf(reqString, envID)
 
-	res, err := c.request("POST", reqUrl, map[string]string{
-		"environment_id": envID,
-		"port":           port,
-		"access":         access,
+	res, err := c.request("POST", reqURL, createDevURLRequest{
+		EnvID:  envID,
+		Port:   port,
+		Access: access,
+		Name:   name,
 	})
+	if res != nil && res.StatusCode == http.StatusConflict {
+		return xerrors.Errorf("Failed to create devurl. Check that the port and name are unique.")
+	}
 	if err != nil {
 		return err
 	}
+	return nil
+}
 
-	if res.StatusCode != http.StatusOK {
-		return bodyError(res)
-	}
```

Comment:
I thought it might be nicer to give a plain English error message rather than just dump the entire HTTP error response to console.

Reply:
Usually the API returns helpful error text and resolution advice. If the responses aren't helpful, maybe we should make them that way?

Reply:
Definitely. I'll look into it. Would be nice if the coder-cli didn't have to do its own explanations and could re-use what we (hopefully!) give back to the web dashboard etc.

Reply:
If it's OK, I'll leave deeper changes to error message to another PR since it'll involve possible changes to cdr/enterprise as well.